python/dolfinx/wrappers/nls.cpp

```diff
@@ -27,17 +27,61 @@ void nls(py::module& m)
       .def(py::init([](const MPICommWrapper comm) {
         return std::make_unique<dolfinx::nls::NewtonSolver>(comm.get());
       }))
-      .def("setF", &dolfinx::nls::NewtonSolver::setF)
-      .def("setJ", &dolfinx::nls::NewtonSolver::setJ)
-      .def("setP", &dolfinx::nls::NewtonSolver::setP)
-      .def("set_form", &dolfinx::nls::NewtonSolver::set_form)
-      .def("solve", &dolfinx::nls::NewtonSolver::solve)
-      .def_readwrite("atol", &dolfinx::nls::NewtonSolver::atol)
-      .def_readwrite("rtol", &dolfinx::nls::NewtonSolver::rtol)
+      .def("setF", &dolfinx::nls::NewtonSolver::setF,
```

Comment:
I'm not keen on extensive docs in the `cpp` wrappers. For me it makes it hard to read the code because the code becomes hard to find amongst the docstrings. I prefer to see it documented from Python.